src/lib.rs

```diff
@@ -68,7 +70,13 @@ pub const PK_SIZE: usize = 48;
 #[cfg(not(feature = "use-insecure-test-only-mock-crypto"))]
 pub const SIG_SIZE: usize = 96;
 
-const ERR_OS_RNG: &str = "could not initialize the OS random number generator";
+fn gen_fr_vec<T: RngCore>(rng: &mut T, len: usize) -> Vec<Fr> {
+    let mut v = Vec::new();
```

Comment:
Maybe `Vec::with_capacity(len)` to avoid reallocation. Or just one line:
```rust
iter::repeat_with(|| Fr::random(rng)).take(len).collect()
```

Reply:
Not bad, but I hope rust can give me something like list comprehension :P